src/ol/interaction/Snap.js

```diff
@@ -114,6 +115,12 @@ class Snap extends PointerInteraction {
 
     super(pointerOptions);
 
+    this.on;
+
+    this.once;
+
+    this.un;
```

Comment:
Types are missing. I'd suggest adding a `SnapOnSignature`, similar to the `Modify` interaction's `ModifyOnSignature`, and using it here as `@type {SnapOnSignature<import("../events").EventsKey>}`

Reply:
Added `SnapOnSignature`